cpp/src/common/cumlHandle_impl.cpp

```diff
@@ -0,0 +1,198 @@
+/*
```

Comment:
It doesn't need to be done in this PR but it would be nice to consolidate all of these files into a couple of even a single file that contains all the handle abstractions and potentially even the implementations.